shared/chat/conversation/input-area/normal/platform-input.native.tsx

```diff
@@ -337,19 +344,44 @@ const Action = React.memo((props: ActionProps) => {
 })
 
 type AudioStarterProps = {
+  dragY: Kb.NativeAnimated.Value
+  locked: boolean
   recording: boolean
   lockRecording: () => void
   enableRecording: () => void
   stopRecording: (st: Types.AudioStopType) => void
 }
 
-const maxAudioDrift = -20
+const maxCancelDrift = -20
+const maxLockDrift = -70
 
 const AudioStarter = (props: AudioStarterProps) => {
   let longPressTimer
+  const locked = React.useRef<boolean>(false)
+  React.useEffect(() => {
+    if (locked.current && !props.locked) {
```

Comment:
you should do `const {locked} = props` so you don't close over props

Reply:
Why? There are like 5 props.